src/cmd/linuxkit/pkglib/build.go

```diff
@@ -95,12 +102,52 @@ func WithBuildTargetDockerCache() BuildOpt {
 // WithBuildCacheDir provide a build cache directory to use
 func WithBuildCacheDir(dir string) BuildOpt {
 	return func(bo *buildOpts) error {
-		bo.cache = dir
+		bo.cacheDir = dir
 		return nil
 	}
 }
 
-// Build builds the package
+// WithBuildPlatforms which platforms to build for
+func WithBuildPlatforms(platforms ...imagespec.Platform) BuildOpt {
+	return func(bo *buildOpts) error {
+		bo.platforms = platforms
+		return nil
+	}
+}
+
+// WithBuildBuilders which builders, as named contexts per platform, to use
+func WithBuildBuilders(builders map[string]string) BuildOpt {
+	return func(bo *buildOpts) error {
+		bo.builders = builders
+		return nil
+	}
+}
+
+// WithBuildDocker provides a docker runner to use. If nil, defaults to the current platform
+func WithBuildDocker(runner dockerRunner) BuildOpt {
+	return func(bo *buildOpts) error {
+		bo.runner = runner
+		return nil
+	}
+}
+
+// WithBuildCacheProvider provides a cacheProvider to use. If nil, defaults to the one shipped with linuxkit
+func WithBuildCacheProvider(c lktspec.CacheProvider) BuildOpt {
+	return func(bo *buildOpts) error {
+		bo.cacheProvider = c
+		return nil
+	}
+}
+
+// WithBuildOutputWriter set the output writer for messages. If nil, defaults to stdout
+func WithBuildOutputWriter(w io.Writer) BuildOpt {
+	return func(bo *buildOpts) error {
+		bo.writer = w
+		return nil
+	}
+}
+
+// BuildNew builds the package
```

Comment:
`./pkglib/build.go:150:1: comment on exported method Pkg.Build should be of the form "Build ..."`